metadata/src/main/java/org/apache/kafka/metadata/PartitionRegistration.java

```diff
@@ -102,6 +112,8 @@ public Builder setPartitionEpoch(Integer partitionEpoch) {
         public PartitionRegistration build() {
             if (replicas == null) {
                 throw new IllegalStateException("You must set replicas.");
+            } else if (directories != null && directories.length != replicas.length) {
+                throw new IllegalStateException("The lengths for replicas and directories do not match.");
```

Comment:
Noting that  we don't throw `InvalidReplicaDirectoriesException` because this is in the Builder as opposed to in an actual record.